cl/scrapers/management/commands/update_from_text.py

```diff
@@ -0,0 +1,163 @@
+from cl.search.models import PRECEDENTIAL_STATUS, Opinion, OpinionCluster
+
+
+def rerun_extract_from_text(
+    opinion: Opinion, juriscraper_module: str, stats: dict[str, int]
+):
+    """
+    Reruns `update_document_from_text` from the scraper flow, saving changes
+
+    `update_document_from_text` calls `Site.extract_from_text` and assigns
+    any changes to the proper objets, in place, but they are not saved.
+    This method saves the ones with actual changes
+
+    :param opinion: the Opinion on which to apply extract_from_text
+    :param juriscraper_module: the scraper module path
+    :param stats: dict to accumulate counts for reporting. Modified in place
+
+    :return None
+    """
+    with transaction.atomic():
+        changes = update_document_from_text(opinion, juriscraper_module)
+        if not changes:
+            logger.info("Did not get any metadata for opinion %s", opinion.id)
+            return
+
+        logger.info("Processing opinion %s", opinion.id)
+
+        # Check if changes exist before saving, to prevent unecessary DB queries
+        if changes.get("Docket"):
+            opinion.cluster.docket.save()
+            logger.debug(
+                "Docket %s updated with data %s",
+                opinion.cluster.docket.id,
+                changes["Docket"],
+            )
+            stats["Docket"] += 1
+
+        if changes.get("OpinionCluster"):
+            opinion.cluster.save()
+            logger.debug(
+                "OpinionCluster %s updated with data %s",
+                opinion.cluster.id,
+                changes["OpinionCluster"],
+            )
+            stats["OpinionCluster"] += 1
+
+        if changes.get("Opinion"):
+            opinion.save()
+            logger.debug("Opinion updated with data %s", changes["Opinion"])
+            stats["Opinion"] += 1
+
+        if changes.get("Citation"):
+            if changes["Citation"].get("citation_created"):
+                logger.info(
+                    "Citation created with data %s", changes["Citation"]
+                )
+                stats["Citation"] += 1
+            else:
+                logger.debug(
+                    "Citation not created. Data %s", changes["Citation"]
+                )
+
+
+class Command(VerboseCommand):
+    help = """Updates objects by running Site.extract_from_text
+    over extracted content found on Opinion.plain_text or Opinion.html.
+
+    If `--opinion-ids` is used, filters will be ignored.
+    If not, the 2 date filters will be required, to prevent triggering
+    unwanted reprocessing of the whole court's dataset
+
+    Recommended use is to run over a sample of the target time period
+    and check if updates over Docket, OpinionCluster, Opinion and
+    Citation are as expected
+    """
+    stats = {}  # assigned at the end of a command run, for testing
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--juriscraper-module",
```

Comment:
Perhaps this isnt possible - but do you think we could get away from the juriscraper module and just use court id?

Reply:
maybe something like 
```
import pkgutil
import importlib
def find_juriscraper_module_from_court_id(court_id: str):
    # Import the base package
    package = importlib.import_module('juriscraper.opinions.united_states')
    for _, modname, ispkg in pkgutil.walk_packages(path=package.__path__,
                                                   prefix=package.__name__ + '.'):
        if modname.endswith(court_id):
            return modname
    return ""

court_id = "wis"
juriscraper_module = find_juriscraper_module_from_court_id(court_id)
```

Reply:
I could be wrong but that would make this a lot easier to process - but perhaps there is a reason we always strictly follow module paths.

Reply:
I think we could support both passing a court_id or a full module; and the court_id argument should error if more than 1 court is found; so the user is forced to use the full module in that case

I think the full module is useful to differentiate between oral arguments and opinion scrapers for the same court, which would be ambiguous if only the court_id was passed. Given that this command should work on opinions only, that ambiguity doesn't count.

On the other hand some court_ids remain ambiguous, even in the context of opinions only. For example `lactapp_1` and `lactapp_5` scrape totally different sites; but their court_id is the same.